tests/v2/validation/provisioning/rke2/etcd_bkp_restore_test.go

```diff
@@ -0,0 +1,504 @@
+	r.watchAndWaitForPods(client, clusterID)
+	logrus.Infof("All pods are up and running.............")
+
+	logrus.Infof("fetching deployment list to validate restore.............")
+	deploymentsList, err := deployments.ListDeployments(client, clusterID, &types.ListOpts{
+		Filters: map[string]interface{}{
+			"fieldSelector": "metadata.namespace=" + r.ns,
+		},
+	})
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), 1, len(deploymentsList.Items))
+	require.Equal(r.T(), w1Name, deploymentsList.Items[0].ObjectMeta.Name)
+	logrus.Infof(" deployment list validated successfully.............")
+
+	logrus.Infof("fetching ingresses list to validate restore.............")
+	ingressesList, err := ingresses.ListIngresses(client, clusterID, &types.ListOpts{
+		Filters: map[string]interface{}{
+			"fieldSelector": "metadata.namespace=" + r.ns,
+		},
+	})
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), 1, len(ingressesList.Items))
+	require.Equal(r.T(), ingress1Name, ingressesList.Items[0].ObjectMeta.Name)
+	logrus.Infof("ingress list validated successfully.............")
+
+	cluster, _, err = clusters.GetProvisioningClusterByName(client, clusterName, r.ns)
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), initialK8sVersion, cluster.Spec.KubernetesVersion)
+}
+
+func (r *RKE2EtcdSnapshotRestoreTestSuite) createRKE2NodeDriverCluster(session *session.Session,
+	client *rancher.Client, provider *Provider, clusterName string, k8sVersion string) (*provisioningV1.SteveAPIObject, error) {
+
+	nodeRoles := []machinepools.NodeRoles{
+		{
+			ControlPlane: true,
+			Etcd:         false,
+			Worker:       false,
+			Quantity:     2,
+		},
+		{
+			ControlPlane: false,
+			Etcd:         true,
+			Worker:       false,
+			Quantity:     3,
+		},
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     3,
+		},
+	}
+
+	cloudCredential, err := provider.CloudCredFunc(client)
+	require.NoError(r.T(), err)
+
+	generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+	machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, r.ns)
+
+	machineConfigResp, err := client.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+	require.NoError(r.T(), err)
+
+	machinePools := machinepools.RKEMachinePoolSetup(nodeRoles, machineConfigResp)
+
+	cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, r.ns, r.cnis[0], cloudCredential.ID, k8sVersion, machinePools)
+
+	return clusters.CreateK3SRKE2Cluster(client, cluster)
+
+}
+
+func (r *RKE2EtcdSnapshotRestoreTestSuite) watchAndWaitForCluster(client *rancher.Client, clusterName string) {
+	logrus.Infof("waiting for cluster to be up.............")
+
+	err := kwait.Poll(5*time.Second, time.Duration(defaults.WatchTimeoutSeconds)*time.Second, func() (done bool, err error) {
+		clusterObj, _, err := clusters.GetProvisioningClusterByName(client, clusterName, r.ns)
+		if err != nil {
+			logrus.Infof(err.Error())
```

Comment:
Will this pollute the log messages too much?

Reply:
Yeah, removed it.